ckanext/spatial/public/js/spatial_query.js

```diff
@@ -197,33 +197,31 @@ this.ckan.module('spatial-query', function ($, _) {
 
     // Is there an existing box from a previous search?
     _setPreviousBBBox: function(map, zoom=true) {
-      previous_bbox = this._getParameterByName('ext_bbox');
+      let module = this;
+      previous_bbox = module._getParameterByName('ext_bbox');
       if (previous_bbox) {
-        $('#ext_bbox').val(previous_bbox);
-        this.extentLayer = this._drawExtentFromCoords(previous_bbox.split(','))
-        map.addLayer(this.extentLayer);
+        module.ext_bbox_input.val(previous_bbox);
+        module.extentLayer = module._drawExtentFromCoords(previous_bbox.split(','))
+        map.addLayer(module.extentLayer);
         if (zoom) {
-          map.fitBounds(this.extentLayer.getBounds(), {"animate": false, "padding": [20, 20]});
+          map.fitBounds(module.extentLayer.getBounds(), {"animate": false, "padding": [20, 20]});
         }
       } else {
-        map.fitBounds(this.options.default_extent, {"animate": false});
+        map.fitBounds(module.options.default_extent, {"animate": false});
       }
-
     },
 
     _onReady: function() {
       let module = this;
       let map;
-      let form = $(".search-form");
-
-      var buttons;
+      let form = $('#dataset-search-form');
+      let bbox_input_id = 'ext_bbox';
 
-      // Add necessary fields to the search form if not already created
-      $(['ext_bbox']).each(function(index, item){
-        if ($("#" + item).length === 0) {
-          $('<input type="hidden" />').attr({'id': item, 'name': item}).appendTo(form);
-        }
-      });
+      // Add necessary field to the search form if not already created
+      if ($("#" + bbox_input_id).length === 0) {
+        $('<input type="hidden" />').attr({'id': bbox_input_id, 'name': bbox_input_id}).appendTo(form);
```

Comment:
Since the `form` was actually 2 forms in our case due to the jquery lookup by class, the `appendTo(form)` was adding the hidden input to both forms. Then the later lookup and value setting (`$('#ext_bbox').val`) only found the first input on the page with that ID and set the val on that. Not the right form in this case.